django/test/selenium.py

```diff
@@ -0,0 +1,53 @@
+import os
+import types
+from unittest import skipIf
+
+from django.test.utils import tag
+from django.utils.module_loading import import_string
+
+
+class SeleniumMetaClass(type):
+
+    def __new__(cls, name, bases, attrs):
+        """
+        Dynamically inject new methods for running tests in different browsers
+        when multiple browser specs are provided (e.g. --selenium=firefox,chrome).
+        """
+        browsers = os.environ.get('DJANGO_SELENIUM_SPECS', '').split(',')
+        if browsers[0]:
+            for key, value in attrs.items():
+                if isinstance(value, types.FunctionType) and key.startswith('test'):
+                    method = value
+                    for index, spec in enumerate(browsers):
+                        # Create new methods for browsers other than first one.
+                        if index > 0:
+                            method_name = '%s__%s' % (key, spec)
+                            method = types.FunctionType(
+                                method.func_code,
+                                method.func_globals,
+                                str(method_name),
+                                method.func_defaults,
+                                method.func_closure,
+                            )
+                            attrs[method_name] = method
+                        setattr(method, 'browser_spec', spec)
+        return type.__new__(cls, name, bases, attrs)
+
+
+skip_selenium = not os.environ.get('DJANGO_SELENIUM_SPECS', '').split(',')[0]
+
+
+@tag('selenium')
+@skipIf(skip_selenium, 'Selenium tests not requested.')
+class SeleniumTestCaseMixin(object):
+
+    __metaclass__ = SeleniumMetaClass
```

Comment:
Ah, need to use https://pythonhosted.org/six/#six.add_metaclass to fix it on Python 3 (that's why I had problems).